### PR TITLE
Stage 2

### PR DESCRIPTION
**Pull request by markovav-official**

### CODE REVIEW

File: output/q2.jpg
Comment (moflotas) [inline output/q2.jpg:1]: are there all states at the bottom?
also, where is manual on the graph?

File: output/q3.jpg
Comment (moflotas) [inline output/q3.jpg:1]: maybe `distribution`, not `counts`?

File: output/q5.jpg
Comment (moflotas) [inline output/q5.jpg:1]: I believe you cannot call it correlation.
Better `Graph of condition dependence on odomoter`

File: scripts/stage2.sh [hunk 1]
```diff
@@ -1 +1,20 @@
...
+
+beeline -u jdbc:hive2://hadoop-03.uni.innopolis.ru:10001 -n team21 -p $PASSWORD -f sql/db.hql > output/hive_results.txt 2> /dev/null
+beeline -u jdbc:hive2://hadoop-03.uni.innopolis.ru:10001 -n team21 -p $PASSWORD -f sql/optimizations.hql > /dev/null 2> /dev/null
+
+beeline -u jdbc:hive2://hadoop-03.uni.innopolis.ru:10001 -n team21 -p $PASSWORD -f sql/q1.hql --hiveconf hive.resultset.use.unique.column.names=false > output/q1.csv 2> /dev/null
+beeline -u jdbc:hive2://hadoop-03.uni.innopolis.ru:10001 -n team21 -p $PASSWORD -f sql/q2.hql --hiveconf hive.resultset.use.unique.column.names=false > output/q2.csv 2> /dev/null
+beeline -u jdbc:hive2://hadoop-03.uni.innopolis.ru:10001 -n team21 -p $PASSWORD -f sql/q3.hql --hiveconf hive.resultset.use.unique.column.names=false > output/q3.csv 2> /dev/null
+beeline -u jdbc:hive2://hadoop-03.uni.innopolis.ru:10001 -n team21 -p $PASSWORD -f sql/q4.hql --hiveconf hive.resultset.use.unique.column.names=false > output/q4.csv 2> /dev/null
+beeline -u jdbc:hive2://hadoop-03.uni.innopolis.ru:10001 -n team21 -p $PASSWORD -f sql/q5.hql --hiveconf hive.resultset.use.unique.column.names=false > output/q5.csv 2> /dev/null
```
Comment (moflotas) [inline scripts/stage2.sh]: extract to a function, no?)

File: sql/create_tables.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+CREATE TABLE car_prices (
+    year INT,
+    make VARCHAR(255),
+    model VARCHAR(255),
+    trim VARCHAR(255),
+    body VARCHAR(255),
+    transmission VARCHAR(255),
+    vin VARCHAR(255),
+    state VARCHAR(255),
```
Comment (moflotas) [inline sql/create_tables.sql]: now, when the data is preprocessed, we can use `VARCHAR(2)` for state
```suggestion
    state VARCHAR(2),
```

File: sql/q1.hql [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    location 'project/hive/warehouse/q1';
+
+-- to not display table names with column names
+SET hive.resultset.use.unique.column.names = false;
+
+INSERT INTO q1_results
+SELECT color, AVG(sellingprice) AS average_price
+FROM car_prices_part_buck
+WHERE color IS NOT NULL
+  AND color != '—'
```
Comment (moflotas) [inline sql/q1.hql]: where should we made the conversion of `-` to `NULL`?

**Review by moflotas — CHANGES_REQUESTED** [review]
Almost fine, but some questions should be discussed

**Review by moflotas — APPROVED** [review]
Cool

**Review by Nertsal — APPROVED** [review]

**Pull request merged**